src/middlewared/middlewared/plugins/pool.py

```diff
@@ -3120,10 +3162,19 @@ def add(attr):
         ('edit', _add_inherit('readonly')),
         ('edit', _add_inherit('recordsize')),
         ('edit', _add_inherit('snapdir')),
+        ('edit', _user_props_rename('user_properties')),
         ('add', Inheritable('quota_warning', value=Int('quota_warning', validators=[Range(0, 100)]))),
         ('add', Inheritable('quota_critical', value=Int('quota_critical', validators=[Range(0, 100)]))),
         ('add', Inheritable('refquota_warning', value=Int('refquota_warning', validators=[Range(0, 100)]))),
         ('add', Inheritable('refquota_critical', value=Int('refquota_critical', validators=[Range(0, 100)]))),
+        ('add', List(
```

Comment:
Won't it be simpler to have `user_properties` schema without the `remove` key and add for update method just add `user_properties_update` schema that has the `remove` key?